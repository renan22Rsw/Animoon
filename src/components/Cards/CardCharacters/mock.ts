interface mockTypes {
  id: number;
  name: string;
  image: string;
  role: string;
  voices: {
    id: number;
    name: string;
    image: string;
    language: string;
  }[];
}

export const mock: mockTypes = {
  id: 1,
  name: "Naruto",
  image: "https://example.com/character-image.jpg",
  role: "Main",
  voices: [
    {
      id: 1,
      name: "Junko Takeuch",
      image: "https://example.com/voices-image.jpg",
      language: "Japanease",
    },
  ],
};
