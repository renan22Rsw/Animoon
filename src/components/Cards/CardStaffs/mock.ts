interface mockTypes {
  id: number;
  name: string;
  image: string;
  occupation: string;
}

export const mock: mockTypes = {
  id: 1,
  name: "Masashi Kishimoto",
  image: "https://example.com/staff-image.jpg",
  occupation: "Original Creator",
};
