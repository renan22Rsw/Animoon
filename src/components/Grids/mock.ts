interface mockAnimesAndMangas {
  id: number;
  title: {
    romaji: string;
  };
  coverImage: {
    large: string;
  };
}

interface mockCharacter {
  id: number;
  name: {
    full: string;
  };
  image: {
    large: string;
  };
}

export const mockAnimesAndMangasMainPage: mockAnimesAndMangas = {
  id: 1,
  title: { romaji: "Akame ga kill" },
  coverImage: { large: "https://example.com/anime-image.jpg" },
};

export const mockCharacterMainPage: mockCharacter = {
  id: 2,
  name: { full: "Akame" },
  image: { large: "https://example.com/character-image.jpg" },
};
