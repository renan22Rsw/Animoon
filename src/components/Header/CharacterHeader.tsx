import Image from "next/image";
import React from "react";
import TextArea from "../TextArea/TextArea";

interface CharacterHeaderProps {
  name: string;
  nativeName: string;
  alternativeName?: string[];
  age: string;
  image: string;
  gender: string;
  month: number;
  day: number;
  bloodType: string;
  hometown?: string;
  yearsActive?: number[];
  description: string;
}

const CharacterHeader = ({
  name,
  nativeName,
  alternativeName,
  age,
  image,
  gender,
  month,
  day,
  bloodType,
  hometown,
  yearsActive,
  description,
}: CharacterHeaderProps) => {
  return (
    <div>
      <div className="bg-[#1C1C1C] pb-[30px] pt-[20px] h-[200px] text-[#9FADB3]  ">
        <div className="max-w-[1200px] hidden m-8 md:flex md:items-center lg:mx-10 2xl:mx-auto">
          <Image
            width={130}
            height={130}
            src={image}
            alt="Anime-Image"
            style={{ height: "345px", width: "230px" }}
            priority={true}
            className="rounded-md "
          />
          <div className="my-4 mx-8  h-[300px]">
            <h2 className="text-[#ADC0D2] text-3xl font-bold py-2">{name}</h2>
            <p className="md:text-sm lg:text-lg">
              {nativeName
                ? nativeName
                : ""
                ? alternativeName
                : alternativeName
                ? nativeName && alternativeName
                : nativeName + "" + alternativeName}
            </p>

            <div className="relative top-[100px]">
              {month || day ? <h4>Birthday: {month + "/" + day}</h4> : ""}

              {age ? <h4>Age: {age}</h4> : ""}

              {gender ? <h4>Gender: {gender} </h4> : ""}

              {yearsActive
                ? yearsActive.map((yearActive, index) => (
                    <h4 key={index}>Years active: {yearActive}</h4>
                  ))
                : ""}
              {hometown ? <h4>Hometown: {hometown} </h4> : ""}
              {bloodType ? <h4>Blood Type: {bloodType}</h4> : ""}
              <TextArea>{description}</TextArea>
            </div>
          </div>
        </div>

        <div className="text-center px-4 md:hidden  ">
          <h2 className="text-[#ADC0D2] text-2xl font-bold py-2">{name}</h2>
          <p>
            {nativeName
              ? nativeName
              : ""
              ? alternativeName
              : alternativeName
              ? nativeName && alternativeName
              : nativeName + "" + alternativeName}
          </p>
          <div className="flex justify-center p-4  my-8">
            <Image
              width={130}
              height={130}
              src={image}
              alt="Anime-Image"
              style={{ height: "345px", width: "230px" }}
              priority={true}
              className="rounded-md "
            />
          </div>
          <div className="text-start mx-4">
            {month || day ? <h4>Birthday: {month + "/" + day}</h4> : ""}
            {age ? <h4>Age: {age}</h4> : ""}
            {gender ? <h4>Gender: {gender} </h4> : ""}
            {yearsActive
              ? yearsActive.map((yearActive, index) => (
                  <h4 key={index}>Years active: {yearActive}</h4>
                ))
              : ""}
            {hometown ? <h4>Hometown: {hometown} </h4> : ""}
            {bloodType ? <h4>Blood Type: {bloodType}</h4> : ""}

            <TextArea>{description}</TextArea>
          </div>
        </div>
      </div>
    </div>
  );
};

export default CharacterHeader;
