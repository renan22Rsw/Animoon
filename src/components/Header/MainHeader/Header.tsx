import TextArea from "@/components/TextArea/TextArea";
import Image from "next/image";
import React from "react";

interface HeaderProps {
  title: string;
  description: string;
  coverImage: string;
}

const Header = ({ title, description, coverImage }: HeaderProps) => {
  return (
    <div className="text-white bg-[#1C1C1C] py-4 lg:flex 2xl:px-20  ">
      <div className=" p-4 md:flex  md:items-center md:w-3/4 lg:w-full lg:flex lg:items-center">
        <Image
          width={300}
          height={300}
          layout="fixed"
          quality={100}
          src={coverImage}
          alt="Header-Image"
          style={{ height: "auto", width: "230px" }}
          priority={true}
          className="rounded-md "
        />

        {/* main versions */}
        <div className=" px-4 hidden md:block xl:w-2/4 ">
          <h1 className="py-4 text-xl">{title}</h1>
          <TextArea>{description}</TextArea>
        </div>
      </div>
      {/* cell versions */}
      <div className="w-4/5 px-4 md:hidden">
        <h1>{title}</h1>
      </div>
    </div>
  );
};

export default Header;
