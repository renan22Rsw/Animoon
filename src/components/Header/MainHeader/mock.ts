interface mockTypes {
  title: string;
  description: string;
  coverImage: string;
}

export const mock: mockTypes = {
  title: "Naruto",
  description:
    "Naruto Uzumaki, a hyperactive and knuckle-headed ninja, lives in Konohagakure, the Hidden Leaf village. Moments prior to his birth, a huge demon known as the Kyuubi, the Nine-tailed Fox, attacked Konohagakure and wreaked havoc. In order to put an end to the Kyuubi's rampage, the leader of the village, the 4th Hokage, sacrificed his life and sealed the monstrous beast inside the newborn Naruto.",
  coverImage: "https://example.com/image.jpg",
};
