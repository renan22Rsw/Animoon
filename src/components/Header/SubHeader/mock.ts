interface mockTypes {
  name: string;
  nativeName: string;
  alternativeName?: string[];
  age: string;
  image: string;
  gender: string;
  month: number;
  day: number;
  bloodType: string;
  hometown: string;
  yearsActive?: number[];
  description: string;
}

export const mock: mockTypes = {
  name: "Naruto",
  nativeName: "Naruto Uzumaki",
  alternativeName: ["Nine-Tails Jinchuuriki", " Number One Unpredictable"],
  age: "17",
  image: "http://example.com/image.jpg",
  gender: "Male",
  month: 10,
  day: 10,
  bloodType: "B",
  hometown: "Saitama Prefecture, Japan",
  yearsActive: [1996],
  description: "Born in Konohagakure, a ninja village hidden in the leaves",
};
