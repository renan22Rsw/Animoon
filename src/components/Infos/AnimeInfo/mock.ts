interface mockTypes {
  format: string;
  duration: number;
  status: string;
  seasonYear: number;
  season: string;
  averageScore: number;
  meanScore: number;
  popularity: number;
  favourites: number;
  source: string;
  genres: string[];
}

export const mock: mockTypes = {
  duration: 24,
  averageScore: 85,
  meanScore: 88,
  favourites: 500,
  format: "TV",
  genres: ["comedy", "horror"],
  popularity: 1000,
  season: "winter",
  seasonYear: 2024,
  source: "Light Novel",
  status: "Airing",
};
