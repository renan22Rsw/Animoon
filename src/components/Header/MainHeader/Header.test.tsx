import { render, screen } from "@testing-library/react";
import Header from "./Header";
import { mock } from "./mock";

describe("MainHeader component", () => {
  it("should render images", () => {
    render(<Header {...mock} />);
    const image = screen.getByRole("img");
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute("width", "130");
    expect(image).toHaveAttribute("src");
    expect(image).toHaveAttribute("height", "130");
    expect(image).toHaveAttribute("fetchpriority", "high");
    expect(image).toHaveStyle({
      width: "230px",
      height: "345px",
    });
  });

  it("should render titles", () => {
    render(<Header {...mock} />);
    const title = screen.getAllByText("Naruto");
    expect(title[0]).toBeInTheDocument();
    expect(title[1]).toBeInTheDocument();
  });

  it("should render description", () => {
    render(<Header {...mock} />);
    const description = screen.getByText(mock.description);
    expect(description).toBeInTheDocument();
  });

  it("should render TextArea component", () => {
    render(<Header {...mock} />);
    const textArea = screen.getByTestId("text-area");
    expect(textArea).toBeInTheDocument();
    expect(textArea).toHaveTextContent(mock.description);
  });

  it("should match a snapshot", () => {
    const { container } = render(<Header {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
