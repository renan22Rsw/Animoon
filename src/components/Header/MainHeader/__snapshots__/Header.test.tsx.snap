// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHeader component should match a snapshot 1`] = `
<div>
  <div
    class="text-white bg-[#1C1C1C] py-4 lg:flex 2xl:px-20  "
  >
    <div
      class=" p-4 md:flex  md:items-center md:w-3/4 lg:w-full lg:flex lg:items-center"
    >
      <img
        alt="Header-Image"
        class="rounded-md "
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="300"
        src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=100"
        srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=384&q=100 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=100 2x"
        style="color: transparent; height: auto; width: 230px;"
        width="300"
      />
      <div
        class=" px-4 hidden md:block xl:w-2/4 "
      >
        <h1
          class="py-4 text-xl"
        >
          Naruto
        </h1>
        <div
          style="display: flex;"
        >
          <textarea
            aria-label="empty textarea"
            class=" w-full text-sm font-normal bg-transparent  leading-normal  resize-none"
            cols="100"
            data-testid="text-area"
            disabled=""
            placeholder="Empty"
            rows="1"
            style="height: 0px; overflow: hidden;"
          >
            Naruto Uzumaki, a hyperactive and knuckle-headed ninja, lives in Konohagakure, the Hidden Leaf village. Moments prior to his birth, a huge demon known as the Kyuubi, the Nine-tailed Fox, attacked Konohagakure and wreaked havoc. In order to put an end to the Kyuubi's rampage, the leader of the village, the 4th Hokage, sacrificed his life and sealed the monstrous beast inside the newborn Naruto.
          </textarea>
          <textarea
            aria-hidden="true"
            class=" w-full text-sm font-normal bg-transparent  leading-normal  resize-none"
            readonly=""
            style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px;"
            tabindex="-1"
          />
        </div>
      </div>
    </div>
    <div
      class="w-4/5 px-4 md:hidden"
    >
      <h1>
        Naruto
      </h1>
    </div>
  </div>
</div>
`;
